src/python/pants/backend/project_info/dependencies.py

```diff
@@ -31,17 +43,74 @@ class DependenciesSubsystem(LineOriented, GoalSubsystem):
         default=False,
         help="Include the input targets in the output, along with the dependencies.",
     )
+    format = EnumOption(
+        default=DependenciesOutputFormat.merged,
+        help="Output format for listing dependencies.",
+    )
 
 
 class Dependencies(Goal):
     subsystem_cls = DependenciesSubsystem
     environment_behavior = Goal.EnvironmentBehavior.LOCAL_ONLY
 
 
-@goal_rule
-async def dependencies(
-    console: Console, addresses: Addresses, dependencies_subsystem: DependenciesSubsystem
-) -> Dependencies:
+async def list_dependencies_as_json(
+    addresses: Addresses, dependencies_subsystem: DependenciesSubsystem, console: Console
+) -> None:
+    """Get dependencies for given addresses and list them in the console in JSON.
+
+    Note that `--closed` option is ignored as it doesn't make sense to duplicate source address in
+    the list of its dependencies.
```

Comment:
This is because a graph is going to be constructed and one wouldn't normally want to have a node in the list of its dependency as this would be creating a cycle, i.e. `{N: [D1, D2, N}`

Reply:
Makes sense. We should add this mention to the `closed` option on the subsystem

Reply:
Or error if closed is specified?

Reply:
Thanks, Benjy, this is, indeed, arguably a better option. Addressed in https://github.com/pantsbuild/pants/pull/20457